OpenEphysBinaryRecordingExtractor is not working with latest Neo update

I believe that after this PR was merged and released in Neo: https://github.com/NeuralEnsemble/python-neo/pull/1624

We are now getting errors in CI like https://github.com/calderast/jdb_to_nwb/actions/runs/14504229897/job/40690639362?pr=120

```
/opt/hostedtoolcache/Python/3.12.10/x64/lib/python3.12/site-packages/jdb_to_nwb/convert_raw_ephys.py:392: in get_raw_ephys_data
    recording = OpenEphysBinaryRecordingExtractor(folder_path=folder_path)
/opt/hostedtoolcache/Python/3.12.10/x64/lib/python3.12/site-packages/spikeinterface/extractors/neoextractors/openephys.py:158: in __init__
    NeoBaseRecordingExtractor.__init__(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <[AttributeError("'OpenEphysBinaryRecordingExtractor' object has no attribute '_recording_segments'") raised in repr()] OpenEphysBinaryRecordingExtractor object at 0x7fb84491bbc0>
stream_id = None, stream_name = None, block_index = None
all_annotations = False, use_names_as_ids = False
neo_kwargs = {'dirname': 'tests/test_data/raw_ephys/2022-07-25_15-30-00', 'experiment_names': None, 'load_sync_channel': False}
kwargs = {'all_annotations': False}
stream_channels = array([('Rhythm_FPGA-100.0', '0', '0'),
       ('Rhythm_FPGA-100.0_ADC', '1', '0')],
      dtype=[('name', '<U64'), ('id', '<U64'), ('buffer_id', '<U64')])
stream_names = ['Rhythm_FPGA-100.0', 'Rhythm_FPGA-100.0_ADC']
stream_ids = ['0', '1']

    def __init__(
        self,
        stream_id: Optional[str] = None,
        stream_name: Optional[str] = None,
        block_index: Optional[int] = None,
        all_annotations: bool = False,
        use_names_as_ids: Optional[bool] = None,
        **neo_kwargs: Dict[str, Any],
    ) -> None:
        """
        Initialize a NeoBaseRecordingExtractor instance.
    
        Parameters
        ----------
        stream_id : Optional[str], default: None
            The ID of the stream to extract from the data.
        stream_name : Optional[str], default: None
            The name of the stream to extract from the data.
        block_index : Optional[int], default: None
            The index of the block to extract from the data.
        all_annotations : bool, default: False
            If True, include all annotations in the extracted data.
        use_names_as_ids : Optional[bool], default: None
            If True, use channel names as IDs. Otherwise, use default IDs.
            In NEO the ids are guaranteed to be unique. Names are user defined and can be duplicated.
        neo_kwargs : Dict[str, Any]
            Additional keyword arguments to pass to the NeoBaseExtractor for initialization.
    
        """
    
        _NeoBaseExtractor.__init__(self, block_index, **neo_kwargs)
    
        kwargs = dict(all_annotations=all_annotations)
        if block_index is not None:
            kwargs["block_index"] = block_index
        if stream_name is not None:
            kwargs["stream_name"] = stream_name
        if stream_id is not None:
            kwargs["stream_id"] = stream_id
        if use_names_as_ids is not None:
            kwargs["use_names_as_ids"] = use_names_as_ids
        else:
            use_names_as_ids = False
    
        stream_channels = self.neo_reader.header["signal_streams"]
        stream_names = stream_channels["name"].tolist()
        stream_ids = stream_channels["id"].tolist()
    
        if stream_id is None and stream_name is None:
            if stream_channels.size > 1:
>               raise ValueError(
                    f"This reader have several streams: \n`stream_names`: {stream_names}\n`stream_ids`: {stream_ids}. \n"
                    f"Specify it from the options above with the 'stream_name' or 'stream_id' arguments"
                )
E               ValueError: This reader have several streams: 
E               `stream_names`: ['Rhythm_FPGA-100.0', 'Rhythm_FPGA-100.0_ADC']
E               `stream_ids`: ['0', '1']. 
E               Specify it from the options above with the 'stream_name' or 'stream_id' arguments

/opt/hostedtoolcache/Python/3.12.10/x64/lib/python3.12/site-packages/spikeinterface/extractors/neoextractors/neobaseextractor.py:208: ValueError
```

--- a/src/jdb_to_nwb/convert_raw_ephys.py
+++ b/src/jdb_to_nwb/convert_raw_ephys.py
@@ -389,7 +389,7 @@ def get_raw_ephys_data(
     # NOTE: We could write our own extractor to handle the relatively simple OpenEphys binary format
     # and surrounding files but it is nice to build on the well-tested code of others when possible.
     # However, we should remember that external code may not be well-maintained or may have bugs of their own.
-    streams = OpenEphysBinaryRecordingExtractor.get_stream_names(folder_path=folder_path)
+    streams, _ = OpenEphysBinaryRecordingExtractor.get_streams(folder_path=folder_path)
     logger.debug(f"Found streams in the OpenEphys binary data: {streams}")
 
     streams_without_adc = [s for s in streams if not s.endswith("_ADC")]
@@ -403,7 +403,10 @@ def get_raw_ephys_data(
     assert all([ch.startswith("CH") for ch in recording_sliced.channel_ids]), \
         (f"Some channels do not start with 'CH': {recording_sliced.channel_ids}")
 
-    logger.debug(f"Found {len(recording_sliced.channel_ids)} Open Ephys channels to convert: {recording_sliced.channel_ids}")
+    logger.debug(
+        f"Found {len(recording_sliced.channel_ids)} Open Ephys channels to convert: "
+        f"{recording_sliced.channel_ids}"
+    )
 
     # Get the channel conversion factor
     channel_conversion_factors_uv = recording_sliced.get_channel_gains()

--- a/src/jdb_to_nwb/convert_raw_ephys.py
+++ b/src/jdb_to_nwb/convert_raw_ephys.py
@@ -389,14 +389,21 @@ def get_raw_ephys_data(
     # NOTE: We could write our own extractor to handle the relatively simple OpenEphys binary format
     # and surrounding files but it is nice to build on the well-tested code of others when possible.
     # However, we should remember that external code may not be well-maintained or may have bugs of their own.
-    recording = OpenEphysBinaryRecordingExtractor(folder_path=folder_path)
+    streams = OpenEphysBinaryRecordingExtractor.get_stream_names(folder_path=folder_path)
+    logger.debug(f"Found streams in the OpenEphys binary data: {streams}")
 
-    # Select only the channels that start with "CH"
+    streams_without_adc = [s for s in streams if not s.endswith("_ADC")]
+    assert len(streams_without_adc) == 1, \
+        (f"More than one non-ADC stream found in the OpenEphys binary data: {streams_without_adc}")
+    
     # Ignore the "ADC" channels
-    channel_ids_to_convert = [ch for ch in recording.channel_ids if ch.startswith("CH")]
-    recording_sliced = recording.select_channels(channel_ids=channel_ids_to_convert)
+    recording_sliced = OpenEphysBinaryRecordingExtractor(folder_path=folder_path, stream_name=streams_without_adc[0])
+    
+    # Confirm all channel names start with "CH"
+    assert all([ch.startswith("CH") for ch in recording_sliced.channel_ids]), \
+        (f"Some channels do not start with 'CH': {recording_sliced.channel_ids}")
 
-    logger.debug(f"Found {len(channel_ids_to_convert)} Open Ephys channels to convert: {channel_ids_to_convert}")
+    logger.debug(f"Found {len(recording_sliced.channel_ids)} Open Ephys channels to convert: {recording_sliced.channel_ids}")
 
     # Get the channel conversion factor
     channel_conversion_factors_uv = recording_sliced.get_channel_gains()
@@ -420,7 +427,7 @@ def get_raw_ephys_data(
     # Get the original timestamps (in seconds)
     original_timestamps = recording_sliced.get_times()
 
-    logger.debug(f"Open Ephys sampling frequency: {recording.get_sampling_frequency()}")
+    logger.debug(f"Open Ephys sampling frequency: {recording_sliced.get_sampling_frequency()}")
 
     # Create a SpikeInterfaceRecordingDataChunkIterator using all default buffering and
     # chunking options. This will be passed to the pynwb.ecephys.ElectricalSeries

